### ISSUE TITLE
RAC: setting `aria-label` on `Section` not working

### ISSUE THREAD

**Issue by tienpv222** [adobe/react-spectrum#5602]
### Provide a general summary of the issue here

Setting `aria-label` on `Section` doesn't apply the attribute.

### 🤔 Expected Behavior?

`aria-label` attribute should be applied on the element.

### 😯 Current Behavior

`aria-label` attribute isn't applied on the element.

### 🖥️ Steps to Reproduce

https://stackblitz.com/edit/stackblitz-starters-t1hrwr?file=src%2FApp.tsx

### Version

1.0.0-rc.0

### What browsers are you seeing the problem on?

Microsoft Edge

### What operating system are you using?

Windows 11 Education - 10.0.22621 Build 22621

**Comment by snowystinger** [comment]
Thanks for reporting, I've opened a PR to address it
